src/com/facebook/buck/jvm/kotlin/ExternalKotlinc.java

```diff
@@ -73,11 +73,11 @@ public ExternalKotlinc(Path pathToKotlinc) {
                 return KotlincVersion.of(output);
               }
             });
+    this.kotlinCompilerVersion = getKotlinCompilerVersion();
   }
 
   /** Returns the Kotlin version, or the path if version is unknown */
-  @AddToRuleKey
-  public String getKotlinCompilerVersion() {
+  private String getKotlinCompilerVersion() {
```

Comment:
If you're running this in the constructor anyways to save to the field, it's probably best to just make this return the field. If it changes during runtime, it's already going to cause issues since we memoize version.get().

Reply:
I made it a private helper function that is only called in the constructor and the function uses only "final" fields to produce a compiler version string. That means there won't be any changes during runtime.